core/bin/external_node/src/main.rs

```diff
@@ -544,6 +544,14 @@ async fn main() -> anyhow::Result<()> {
         ([0, 0, 0, 0], config.required.healthcheck_port).into(),
         app_health.clone(),
     );
+    // Start scraping Postgres metrics before store initialization as well.
+    let metrics_pool = connection_pool.clone();
+    let mut task_handles = vec![tokio::spawn(async move {
+        metrics_pool
+            .run_postgres_metrics_scraping(Duration::from_secs(60))
+            .await;
+        Ok(())
+    })];
```

Comment:
it's now only run for EN, we need to make sure we include it when we run nodes via yaml files. Will it be a separate component? @popzxc curious how'd you build id into the Framework

Reply:
Well, it certainly will be a task. Where would it be added? A lot of options, e.g.
1. A dedicated layer that fetches the pool as a resource and spawns a task. Maybe parametrized for a pool type.
2. Can be spawned by the pools layer directly if we want. Maybe in a form `PoolLayer::new().export_extra_metrics(true).build()`
3. Have an umbrella layer for the observability stuff (e.g. we have other tasks that mostly exist for metrics -- we can either spawn them all together or none).

Depends on how we see the use cases.

Reply:
It's run on the main node as well, btw, as a part of the house keeper component.